### PR TITLE
items: detect mode in animation command SFX

### LINKED ISSUE #999: Add water splash sound if Lara runs in 1 water like in TR2+
Requested from custom builders on Discord. I think this would require a global SFX injection.

### PR DESCRIPTION
**Pull request by lahm86**
Resolves #999.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

This allows animation SFX commands to specify whether a sound plays on water, on land or both. Both is the default and will apply to all OG game sounds; builders can use the same flags as defined in TR2+ to set for land/water only.

I was going to introduce a new enum for the mode, but instead opted for the switch statement to convert to the existing `SOUND_PLAY_MODE`. That switch statement could become one line, as follows, but I felt it was more readable like this - but let me know what you think.

`SOUND_PLAY_MODE mode = (((command[1] & 0xC000) >> 14) + 2) % 3;`

The update to `lara_animations.bin` injects some "wet feet" SFX and updates her animation commands to play these in (one-click) water rather than her normal footsteps, as well as some splash sounds in various places. Below is a showcase of this.

https://youtu.be/yt4y5ek8hFc

### CODE REVIEW

File: src/game/items.c [hunk 1]
```diff
@@ -637,6 +645,36 @@ bool Item_GetAnimChange(ITEM_INFO *item, ANIM_STRUCT *anim)
...
 }
 
+void Item_PlayAnimSFX(ITEM_INFO *item, int16_t *command, uint16_t flags)
+{
+    if (item->frame_number != command[0]) {
+        return;
+    }
+
+    SOUND_PLAY_MODE mode;
+    switch (command[1] & 0xC000) {
```
Comment (walkawayy) [inline src/game/items.c]: Should we use constants for `0xC000`, `0x4000`, and `0x8000`?

File: src/game/items.c [hunk 2]
```diff
@@ -637,6 +645,36 @@ bool Item_GetAnimChange(ITEM_INFO *item, ANIM_STRUCT *anim)
...
+
+    if (mode != SPM_ALWAYS) {
+        int16_t height = Item_GetWaterHeight(item);
+        if ((mode == SPM_UNDERWATER && (height >= 0 || height == NO_HEIGHT))
+            || (mode == SPM_NORMAL && height < 0 && height != NO_HEIGHT)) {
+            return;
+        }
+    }
+
+    Sound_Effect(command[1] & 0x3FFF, &item->pos, flags);
```
Comment (walkawayy) [inline src/game/items.c]: `0x3FFF` constant?

**Review by walkawayy — CHANGES_REQUESTED** [review]
All my testing looked good! Just some code cosmetic comments.

File: src/game/items.c (hunk 1 above)
Comment (lahm86) [reply src/game/items.c]: Thanks 👍 I've tidied it up a bit, and managed to get rid of the switch statement in the process.

**Review by walkawayy — APPROVED** [review]

**Review by rr- — APPROVED** [review]

**Pull request merged**